### ISSUE TITLE
RF02: Unqualified reference false positive on LATERAL FLATTEN Snowflake

### ISSUE THREAD

**Issue by dobsontom** [sqlfluff/sqlfluff#6754]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff throws unqualified references on select items in a selection with a single source where the source has undergone a LATERAL FLATTEN.

### Expected Behaviour

No flag on these items.

### Observed Behaviour

The following returns two RF06 unqualified reference errors:

```
select shipment_id
from regen_shipment_info,
    lateral flatten(input => order_id_array)
```

### How to reproduce

Run SQLFluff lint on the above.

### Dialect

Snowflake

### Version

`dbt-core` 1.9.3
`dbt-snowflake` 1.9.2
`sqlfluff` 3.3.1
`sqlfluff-templater-dbt` 3.3.1

### Configuration

```
[sqlfluff]
dialect = snowflake
templater = dbt
max_line_length = 120
large_file_skip_byte_limit = 0

processes = 0

exclude_rules = RF03, RF04, RF06, ST06, ST07

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.identifiers]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by moreaupascal56** [comment]
Hi I understand that this example works on Snowflake, but I am unsure if this needs a fix or not.
Indeed in theory you could have ambiguous column name (if one is named value in the source table) even with only one source. 

I guess it depends on preferences but IMO qualifying both these reference would be better/clearer so I would stick with the current configuration. 

Basically you are saying that in this configuration (1 source, 1 lateral flatten) we do not need to qualify because we have to use the value keyword to get data from the lateral?

Regards
